karapace/backup/topic_configurations.py

```diff
@@ -2,25 +2,12 @@
 Copyright (c) 2023 Aiven Ltd
 See LICENSE for details
 """
-from enum import Enum
-from kafka import KafkaAdminClient
-from kafka.admin import ConfigResource, ConfigResourceType
-from kafka.errors import for_code
-from kafka.protocol.admin import DescribeConfigsRequest
-from typing import Container, Dict, Final
+from __future__ import annotations
 
+from karapace.kafka_admin import ConfigSource, KafkaAdminClient
+from typing import Container, Final
 
-class ConfigSource(int, Enum):
-    UNKNOWN = 0
-    TOPIC_CONFIG = 1
-    DYNAMIC_BROKER_CONFIG = 2
-    DYNAMIC_DEFAULT_BROKER_CONFIG = 3
-    STATIC_BROKER_CONFIG = 4
-    DEFAULT_CONFIG = 5
-    DYNAMIC_BROKER_LOGGER_CONFIG = 6
-
-
-ALL_CONFIG_SOURCES: Final = {item.value for item in ConfigSource.__members__.values()}
+ALL_CONFIG_SOURCES: Final = ConfigSource.__members__.values()
```

Comment:
```suggestion
ALL_CONFIG_SOURCES: Final = ConfigSource
```

nitpick (non-blocking): Code golfing 😉 This constant seems superfluous as enum classes are themselves valid iterables and containers of their members.